responsibleai/responsibleai/modelanalysis/model_analysis.py

```diff
@@ -117,6 +133,120 @@ def __init__(self, model, train, test, target_column,
+            except Exception:
+                raise UserConfigValidationException(
+                    'The serializer should be serializable via pickle')
+
+        if isinstance(train, pd.DataFrame) and isinstance(test, pd.DataFrame):
+            if len(set(train.columns) - set(test.columns)) != 0 or \
+                    len(set(test.columns) - set(train.columns)):
+                raise UserConfigValidationException(
+                    'The features in train and test data do not match')
+
+            if target_column not in list(train.columns) or \
+                    target_column not in list(test.columns):
+                raise UserConfigValidationException(
+                    'Target name {0} not present in train/test data'.format(
+                        target_column)
+                )
+
+            if categorical_features is not None and \
+                    len(categorical_features) > 0:
+                if target_column in categorical_features:
+                    raise UserConfigValidationException(
+                        'Found target name {0} in '
+                        'categorical feature list'.format(
+                            target_column)
+                    )
+
+                if not set(categorical_features).issubset(set(train.columns)):
+                    raise UserConfigValidationException(
+                        'Found some feature names in categorical feature which'
+                        ' do not occur in train data'
+                    )
+
+            if train_labels is not None and task_type == \
+                    ModelTask.CLASSIFICATION:
+                if len(set(train[target_column].unique()) -
+                       set(train_labels)) != 0 or \
+                        len(set(train_labels) -
+                            set(train[target_column].unique())) != 0:
+                    raise UserConfigValidationException(
+                        'The train labels and distinct values in '
+                        'target (train data) do not match')
+
+                if len(set(test[target_column].unique()) -
+                       set(train_labels)) != 0 or \
+                        len(set(train_labels) -
+                            set(test[target_column].unique())) != 0:
+                    raise UserConfigValidationException(
+                        'The train labels and distinct values in '
+                        'target (test data) do not match')
+
+            if model is not None:
+                # Run predict of the model
+                try:
+                    small_train_data = train.iloc[0:1].drop(
+                        [target_column], axis=1)
+                    small_test_data = test.iloc[0:1].drop(
+                        [target_column], axis=1)
+                    model.predict(small_train_data)
+                    model.predict(small_test_data)
+                except Exception:
+                    raise UserConfigValidationException(
```

Comment:
When is this expected?

Reply:
Since almost RAI components rely on predcit()/predict_proba() so adding a check that the predict() works on the train and test data. I should get predeict_proba() as well.

Reply:
Added validation for predcit_proba()